### PR TITLE
Desktop code merged into develop

### LINKED ISSUE #5083: [MERGE] Create a temp branch for desktop->develop merge
**Problem**:
It is important to have a testing ground for the desktop->develop merge.

**Solution**:
Create a branch containing a version of `desktop` rebased against `develop`.

### PR DESCRIPTION
**Pull request by vkjr**
fixes #5083

### Summary:

This idea behind this PR is to merge mobile and desktop code and have it in the same branch. `desktop_develop_merge` branch contains code from `desktop` rebased on `develop`.

In most files code difference handled by `(if platform/desktop? ...)` statements. 

`vector_icons.cljs` works quite differently for desktop. It doesn't use `react-native-svg` library and able to display svg directly in `Image` component (thanks to underlying Qt).

`js_dependencies.cljs` also has a lot of changes between desktop and mobile versions. Mostly because not all 3rdparty dependencies are implemented and needed by desktop build.

The most problematic are files: `VERSION`, `package.json` and `package-lock.json`. They are different for mobile and desktop and `(if desktop?)` doesn't work there...
I saved different versions of these files in `mobile_files` and `desktop_files` folders. 
If you want to build code of mobile app, you should invoke `scripts/prepare-for-platform.sh mobile`. `scripts/prepare-for-platform.sh desktop` for desktop build.
That script will create in root folder links pointing to `package.json`, `VERSION`, `package-lock.json` in `mobile_files` or `desktop_files` directory.

### Review notes (optional):
Here are steps to run `ios` app from the code:

```
scripts/prepare-for-platform.sh mobile
npm install
mvn -f modules/react-native-status/ios/RCTStatus dependency:unpack
ln -sf './node_modules/re-natal/index.js' './re-natal'
(In terminal 1:) make startdev-ios-simulator
(In terminal 2:) react-native run-ios
```

Here are steps to run `desktop` app from the code:

```
scripts/prepare-for-platform.sh desktop
npm install;
ln -sf './node_modules/re-natal/index.js' './re-natal'
./re-natal use-figwheel;
./re-natal enable-source-maps;
(In terminal 1:) lein figwheel-repl desktop (wait until compilation finished)
(In terminal 2:) npm start
(In terminal 3:) ./ubuntu-server.js
(In terminal 3:) react-native run-desktop
```

### Steps to test:
- Open mobile or desktop Status
- Check that it works correctly
- Step 3, etc.

status: ready

### CODE REVIEW

**Comment by vkjr** [comment]
@churik, I recreated PR, could you please test it? Jenkins build passed for android, ios, mac, linux.

@cammellos, @siphiuel, @MaxRis, could you please approve this PR? It has no changes in comparison with the previous one. I simply merged all commits in one because due to a lot of new commits in `develop` there were issues with rebasing on Jenkins

**Review by vitvly — APPROVED** [review]

**Review by maxhora — APPROVED** [review]

**Review by cammellos — APPROVED** [review]

**Comment by churik** [comment]
@vkjr IOS and Android builds are crashed on start;
* TF session: https://app.testfairy.com/projects/4803622-status/builds/8531767/sessions/4396754911/?accessToken=HICLREIFz-zm42NBC9iHFr3Nwts
```
com.facebook.react.common.JavascriptException: Requiring unknown module "react-native-i18n"., stack:
d@2:595
n@2:409
t@2:262
@12:2215284
d@2:768
n@2:339
```

Desktop Mac Osx build starts with error:
image

**Comment by vkjr** [comment]
@churik, new builds available. Please take a look :)

**Comment by churik** [comment]
Test results for testing  IOS and ANdroid builds are [here](https://ethstatus.testrail.net/index.php?/runs/view/887&group_by=cases:section_id&group_order=as)
Tested on Mac OSx build:
- creating account
- adding new contact 
- chatting (1-1, public chats)
- share my public code
- log in\log out

File: src/status_im/ui/screens/browser/db.cljs.orig [hunk 1]
```diff
@@ -0,0 +1,41 @@
+(ns status-im.ui.screens.browser.db
```
Comment (yenda) [inline src/status_im/ui/screens/browser/db.cljs.orig]: failed merge ?

File: src/status_im/ui/components/icons/vector_icons.cljs [hunk 2]
```diff
@@ -2,121 +2,198 @@
...
+            [status-im.ui.components.react :as react]
             [status-im.utils.platform :as platform]
             [status-im.ui.components.styles :as styles]
-            [status-im.ui.components.react :as react]
             [status-im.react-native.js-dependencies :as js-dependencies])
   (:refer-clojure :exclude [use]))
 
-(defn get-property [name]
-  (object/get js-dependencies/svg name))
+(if-not platform/desktop?
```
Comment (yenda) [inline src/status_im/ui/components/icons/vector_icons.cljs]: `when-not`, then no need to wrap statement in `do`

File: src/status_im/ui/components/icons/vector_icons.cljs [hunk 3]
```diff
@@ -2,121 +2,198 @@
...
-            :icons/network             (components.svg/slurp-svg "./resources/icons/network.svg")
-            :icons/wnode               (components.svg/slurp-svg "./resources/icons/wnode.svg")
-            :icons/refresh             (components.svg/slurp-svg "./resources/icons/refresh.svg")
-            :icons/newchat             (components.svg/slurp-svg "./resources/icons/newchat.svg")
-            :icons/logo                (components.svg/slurp-svg "./resources/icons/logo.svg")
-            :icons/camera              (components.svg/slurp-svg "./resources/icons/camera.svg")
-            :icons/check               (components.svg/slurp-svg "./resources/icons/check.svg")
-            :icons/dots                (components.svg/slurp-svg "./resources/icons/dots.svg")
-            :icons/warning             (components.svg/slurp-svg "./resources/icons/warning.svg")})
+(if platform/desktop?
```
Comment (yenda) [inline src/status_im/ui/components/icons/vector_icons.cljs]: wouldn't this work and avoid duplication ?

`(def slurp-svg (if plateform/desktop? js/require components.svg/slurp-svg))`
then use slurp-svg

**Comment by yenda** [comment]
Should this one be closed since there is now https://github.com/status-im/status-react/pull/5266 ?

**Comment by vkjr** [comment]
@yenda, yes, thanks.

**Pull request closed without merging**